File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.py
def solution(n, arr1, arr2):
    answer = []
    
    for i in range(len(arr1)):
        arr1[i] = list(format(arr1[i], 'b').zfill(n))
        arr2[i] = list(format(arr2[i], 'b').zfill(n))
        
    for i in range(len(arr1)):
        arr = []
        for j in range(len(arr2)):
            if arr1[i][j] == '1' or arr2[i][j] == '1':
                arr.append("#")
            else :
                arr.append(" ")
                
        answer.append(''.join(arr))
            
    return answer

File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.js
function solution(n, arr1, arr2) {
    let arr = [];
    
    for(let i = 0; i<n; i++){
        arr.push((arr1[i] | arr2[i]).toString(2).padStart(n, 0));
    }
    
    arr = arr.map(e => e.replace(/1|0/g, num => +num ? '#' : ' '));
    
    return arr;
}